File: python/event.py

class Event:
    def __init__(self, eventType, dt, psk, pri, oid=0, client='', broker='', isBuy=False, vol=0):
        self.__et = eventType
        self.__dt = dt
        self.__psk = psk
        self.__pri = pri
        self.__oid = oid
        self.__cli = client
        self.__bro = broker
        self.__isb = isBuy
        self.__vol = vol
        
    @property
    def eventType(self):
        return self.__et
        
    @property        
    def dateTime(self):
        return self.__dt
        
    @property
    def parsekey(self):
        return self.__psk
        
    @property
    def price(self):
        return self.__pri

    @property
    def orderId(self):
        return self.__oid

    @property
    def client(self):
        return self.__cli

    @property
    def broker(self):
        return self.__bro

    @property
    def isBuy(self):
        return self.__isb
        
    @property
    def volume(self):
        return self.__vol

File: golang/src/events/event.go
package event

import (
    tm "time"
)

type EVENT_TYPE uint8
const (
    _              =   iota
    EVENT_ORDER
    EVENT_EXECUTE
    EVENT_PRICE
)

type Event struct{
    EventType EVENT_TYPE
    OrderId int
    EventTime tm.Time
    Client string
    Broker string
    IsBuy bool
    Security string
    Volume int
    Price float64
}

func InitPrice(dt *tm.Time, sec string, pri float64) (Event,bool){
    return Event{
        EventType: EVENT_PRICE,
        EventTime: *dt,
        Security:  sec,
        Price:     pri,
    },true
}
func InitOrder(oid int, dt *tm.Time, cli , bro string, ib bool, sec string, vol int, pri float64) (Event,bool){
    return Event{
        EventType: EVENT_ORDER,
        OrderId:   oid,
        EventTime: *dt,
        Client:    cli,
        Broker:    bro,
        IsBuy:     ib,
        Security:  sec,
        Volume:    vol,
        Price:     pri,
    },true
}
func InitExecute(oid int, dt *tm.Time, cli , bro string, ib bool, sec string, vol int, pri float64) (Event,bool){
    return Event{
        EventType: EVENT_EXECUTE,
        OrderId:   oid,
        EventTime: *dt,
        Client:    cli,
        Broker:    bro,
        IsBuy:     ib,
        Security:  sec,
        Volume:    vol,
        Price:     pri,
    },true
}
